sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4296,15 +4296,23 @@ pi_result piProgramBuild(pi_program Program, pi_uint32 NumDevices,
   ze_device_handle_t ZeDevice = DeviceList[0]->ZeDevice;
   ze_context_handle_t ZeContext = Program->Context->ZeContext;
   ze_module_handle_t ZeModule = nullptr;
-  ZE_CALL(zeModuleCreate, (ZeContext, ZeDevice, &ZeModuleDesc, &ZeModule,
-                           &Program->ZeBuildLog));
-
+  ze_result_t ZeResult =
+      ZE_CALL_NOCHECK(zeModuleCreate, (ZeContext, ZeDevice, &ZeModuleDesc,
+                                       &ZeModule, &Program->ZeBuildLog));
+  if (ZeResult != ZE_RESULT_SUCCESS) {
+    // We need to clear Program state to avoid double destroy of zeModule in
+    // case where SYCL RT calls piProgramRelease().
+    // We should not return with an error code here due to the comments below.
```

Comment:
I can't understand the comment. It should just say that we need to adjust the PI program such that it can properly be cleaned in piProgramRelease

Reply:
I think the right approach is to clean up the ZeModule we created for the failed program build here in piProgramBuild().
Currently, SYCL RT won't call piProgramRelease() if piProgramBuild returns an error code.
I understand that the user may catch the exception thrown by SYCL RT, and RT may call piProgramRelease().
But, since we set Program->ZeModule = nullptr, L0 won't call zeDestroyModule.